server/src/test/java/org/opensearch/index/translog/transfer/BaseTranslogTransferManagerTests.java

```diff
@@ -294,62 +320,48 @@ public void onUploadFailed(TransferSnapshot transferSnapshot, Exception ex) {
             assertTrue(exception.get() instanceof TranslogUploadFailedException);
             assertEquals("Failed to upload test-to-string", exception.get().getMessage());
         });
-        uploadThread.get().interrupt();
+        uploadThreadList.forEach(Thread::interrupt);
     }
 
     private TransferSnapshot createTransferSnapshot() {
         return new TransferSnapshot() {
             @Override
-            public Set<TransferFileSnapshot> getCheckpointFileSnapshots() {
-                try {
-                    return Set.of(
-                        new CheckpointFileSnapshot(
-                            primaryTerm,
-                            generation,
-                            minTranslogGeneration,
-                            createTempFile(Translog.TRANSLOG_FILE_PREFIX + generation, Translog.CHECKPOINT_SUFFIX),
-                            null
-                        ),
-                        new CheckpointFileSnapshot(
-                            primaryTerm,
-                            generation,
-                            minTranslogGeneration,
-                            createTempFile(Translog.TRANSLOG_FILE_PREFIX + (generation - 1), Translog.CHECKPOINT_SUFFIX),
-                            null
-                        )
-                    );
-                } catch (IOException e) {
-                    throw new AssertionError("Failed to create temp file", e);
-                }
+            public TranslogTransferMetadata getTranslogTransferMetadata() {
```

Comment:
in general I have noticed through the PR that while modifying the classes we are disturbing the order of the methods from its previous state. This makes the review difficult. I would urge you to maintain the same order so that it is clear what is being removed, what is being added and what is being modified.

Reply:
Ack.